pkg/addon/addon.go

```diff
@@ -1092,6 +1085,23 @@ func checkDependencyNeedInstall(ctx context.Context, k8sClient client.Client, de
 		needInstallAddonDep = true
 		depClusters = addonClusters
 	} else {
+		// if addon clusters is nil, meaning to deploy to all clusters
+		if addonClusters == nil {
+			hasTopologyPolicy := false
+			topologyPolicyValue := map[string]interface{}{}
+			for _, policy := range depApp.Spec.Policies {
+				if policy.Type == "topology" {
+					hasTopologyPolicy = true
+					unmarshalErr := json.Unmarshal(policy.Properties.Raw, &topologyPolicyValue)
+					if unmarshalErr != nil {
+						return true, nil, unmarshalErr
```

Comment:
If an error is generated, first return value is useless, please keep it as default, which is `false`.